packages/lodestar/src/network/reqResp.ts

```diff
@@ -117,56 +105,44 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
   public async beaconBlocksByRange(
     peerInfo: PeerInfo,
     request: BeaconBlocksByRangeRequest
-  ): Promise<BeaconBlocksByRangeResponse> {
-    return await this.sendRequest<BeaconBlocksByRangeResponse>(peerInfo, Method.BeaconBlocksByRange, request);
+  ): Promise<SignedBeaconBlock[]> {
+    return await this.sendRequest<SignedBeaconBlock[]>(peerInfo, Method.BeaconBlocksByRange, request);
   }
   public async beaconBlocksByRoot(
     peerInfo: PeerInfo,
     request: BeaconBlocksByRootRequest
-  ): Promise<BeaconBlocksByRootResponse> {
-    return await this.sendRequest<BeaconBlocksByRootResponse>(peerInfo, Method.BeaconBlocksByRoot, request);
+  ): Promise<SignedBeaconBlock[]> {
+    return await this.sendRequest<SignedBeaconBlock[]>(peerInfo, Method.BeaconBlocksByRoot, request);
   }
 
-  private async handleRpcRequestStream(
-    peerId: PeerId, method: Method, source: Promise<Buffer | {slice: () => Buffer}>[]): Promise<ResponseChunk[]> {
-    const result: ResponseChunk[] = [];
-    for await (const val of source) {
-      const data = Buffer.isBuffer(val) ? val : val.slice();
-      const response = await this.handleRequest(peerId, method, data);
-      if (!response.err && (method === Method.BeaconBlocksByRange || method === Method.BeaconBlocksByRoot)) {
-        const blocks = response.output as SignedBeaconBlock[];
-        const chunkResponses = blocks.map(block => ({output: block}));
-        result.push(...(chunkResponses));
-      } else {
-        result.push(response as ResponseChunk);
-      }
-    }  
-    return result;
-  }
-
-  private transformResponse(
-    method: Method,
-    source: Promise<{[Symbol.asyncIterator]: () => AsyncIterator<ResponseChunk>}>
-  ): AsyncIterator<Buffer> {
-    const config = this.config;
-    return (async function * () {
-      const sourceVal = await source;
-      for await (const val of sourceVal) {
-        // each yield result will be sent to stream.sink immediately
-        yield encodeResponseChunk(config, method, val);
-      }
-    })();
+  private handleRpcRequests(
+    peerId: PeerId, method: Method
+  ): ((source: AsyncIterable<Buffer|BufferList>) => AsyncGenerator<ResponseChunk>) {
+    const getResponse = this.getResponse;
+    return (source: AsyncIterable<Buffer|BufferList>) => {
+      return (async function * () {
+        for await (const val of source) {
+          const data = Buffer.isBuffer(val) ? val : val.slice();
+          const response = await getResponse(peerId, method, data);
+          if(response.length) {
+            yield* response;
```

Comment:
This `yield*` is nice :)

Reply:
you learn those tricks if you use react with saga :smile: